core/src/main/java/com/sequenceiq/cloudbreak/service/altus/AltusIAMService.java

```diff
@@ -59,7 +59,18 @@ public void clearFluentMachineUser(Stack stack, Telemetry telemetry) {
 
     // for datalake metering is not supported/required right now
     private boolean isMeteringOrDeploymentReportingSupported(Stack stack, Telemetry telemetry) {
-        return telemetry != null && (telemetry.isReportDeploymentLogs() || (telemetry.isMeteringEnabled() && !StackType.DATALAKE.equals(stack.getType())));
+        return telemetry != null && (isReportDeploymentLogsFeatureEnabled(telemetry) || (isMeteringFeatureEnabled(telemetry)
+                && !StackType.DATALAKE.equals(stack.getType())));
+    }
+
+    private boolean isReportDeploymentLogsFeatureEnabled(Telemetry telemetry) {
+        return telemetry.getFeatures() != null && telemetry.getFeatures().getReportDeploymentLogs() != null
+                && telemetry.getFeatures().getReportDeploymentLogs().isEnabled();
+    }
+
+    private boolean isMeteringFeatureEnabled(Telemetry telemetry) {
```

Comment:
And this could be public methods of the Telemetry class, and could be unit tested then.